client/my-sites/themes/themes-selection.jsx

```diff
@@ -101,6 +106,16 @@ const ThemesSelection = React.createClass( {
 		this.props.onScreenshotClick && this.props.onScreenshotClick( theme );
 	},
 
+	addVerticalToFilters() {
+		const { vertical, filter } = this.props;
+		const result = [];
+
+		filter && result.push( filter );
+		vertical && result.push( vertical );
```

Comment:
I've recently started to appreciate lodash's [`compact`](https://lodash.com/docs/4.16.2#compact) for filtering falsey elements from a list:

```js
const result = [ filter, vertical ];

return compact( result ).join( ',' );
```

(You could even inline the first line.)